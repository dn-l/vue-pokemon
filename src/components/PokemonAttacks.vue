<template>
  <div v-if="attacks">
    <div v-for="(attacks, category) in attacks" :key="category">
      <div v-if="Array.isArray(attacks) && attacks.length" class="my-2">
        <h2 class="text-capitalize">{{ category }} attacks</h2>
        <v-simple-table>
          <thead>
            <tr>
              <th width="50%">Name</th>
              <th>Type</th>
              <th>Damage</th>
            </tr>
          </thead>
          <tbody>
            <tr v-for="attack in attacks" :key="attack.name">
              <td width="50%">{{ attack.name }}</td>
              <td>{{ attack.type }}</td>
              <td>{{ attack.damage }}</td>
            </tr>
          </tbody>
        </v-simple-table>
      </div>
    </div>
  </div>
  <div v-else>
    <h2>No attacks</h2>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  props: ['attacks'],
});
</script>
