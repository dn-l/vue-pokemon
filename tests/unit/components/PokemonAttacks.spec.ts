import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PokemonAttacks from '@/components/PokemonAttacks.vue';

const localVue = createLocalVue();
Vue.use(Vuetify);

const attacks = {
  fast: [{
    name: 'Punch',
    damage: 1,
    type: 'Simple',
  }],
  special: [{
    name: 'Fireball',
    damage: 100,
    type: 'Magical',
  }],
};

describe('PokemonAttacks.vue', () => {
  const wrapper = shallowMount(PokemonAttacks, {
    propsData: attacks,
    localVue,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
