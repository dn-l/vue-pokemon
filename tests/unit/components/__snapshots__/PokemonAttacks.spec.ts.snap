// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonAttacks.vue matches snapshot 1`] = `
<div fast="[object Object]" special="[object Object]">
  <h2>No attacks</h2>
</div>
`;
